packages/forms-web-app/src/controllers/examination/email.js

```diff
@@ -2,24 +2,67 @@ const {
 	routesConfig: {
 		examination: {
 			directory: examinationDirectory,
-			pages: { nameMyself, email }
+			pages: {
+				email,
+				checkYourAnswers: { route: checkYourAnswersRoute }
+			}
 		}
 	}
 } = require('../../routes/config');
 
+const config = require('../../config');
+const examinationSessionStorage = config?.sessionStorage?.examination;
+
 const pageData = {
-	backLinkUrl: `${examinationDirectory + nameMyself.route}`,
 	id: email.id,
 	pageTitle: email.name,
-	title: email.name
+	title: email.name,
+	hint: "We'll use your email address to confirm we've received your submission. We will not publish your email address."
 };
 
 const getEmail = async (req, res) => {
+	const requestSession = req.session;
+	const examinationSession = requestSession?.[examinationSessionStorage.name];
+	const currentView = requestSession?.currentView;
+
+	if (!examinationSession || !currentView || !currentView?.route)
+		return res.status(404).render('error/not-found');
+
+	const userEmail = examinationSession?.userEmail;
+	userEmail && (pageData.email = userEmail);
+
+	pageData.backLinkUrl = `${examinationDirectory + currentView.route}`;
 	res.render(email.view, pageData);
 };
 
 const postEmail = async (req, res) => {
-	res.render(email.view, pageData);
+	const { body = {} } = req;
+	const { errors = {}, errorSummary = [] } = body;
+	const requestSession = req?.session;
+	const currentView = requestSession?.currentView;
+	const examinationStorage = requestSession?.[examinationSessionStorage.name];
+
+	if (!requestSession || !currentView || !examinationStorage)
+		return res.status(404).render('error/not-found');
+
+	examinationStorage?.userEmail && (pageData.email = examinationStorage?.userEmail);
+
+	examinationStorage.userEmail = examinationSessionStorage.property.email = body[email.id];
```

Comment:
I think this should be `examinationStorage.userEmail = examinationStorage[examinationSessionStorage.property.email] = body[email.id];`

`.userEmail` should also be extracted from the config variables

Reply:
Probably what we want is examinationStorage.email

this: examinationSessionStorage.property.email will be transformed in the email value but once it gets reassigned it can override other variables

examinationStorage  = req.session.examination

examinationSessionStorage = config.sessionStorage.examination

examinationStorage[examinationSessionStorage.property.email] = sessionStorage.examination.email (this is wrong)